packages/headless/src/database/mongodb/cart/index.ts

```diff
@@ -0,0 +1,115 @@
+import { Cart, Item } from '../../../entity/cart';
+import { ICartDatabase } from '../../../modules/cart/repositories/cart.database.interface';
+import { CartModel } from './cart.model';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class CartDatabase implements ICartDatabase {
+  //   constructor() {}
+
+  async addItemByUserId(userId: string, item: Item): Promise<Cart | null> {
```

Comment:
Dont need to mention UserId. Item always will be added by user. so keep function name simple `addItem`